core/src/ldml/ldml_processor.cpp

```diff
@@ -438,6 +456,11 @@ ldml_processor::emit_marker(km_core_state *state, KMX_DWORD marker_no) {
   state->context().push_marker(marker_no);
 }
 
+void ldml_processor::emit_invalidate_passthrough_keystroke(km_core_state *state) {
+  state->actions().push_invalidate_context();
+  state->actions().push_emit_keystroke();
+}
+
```

Comment:
encapsulating this in case it needs to change later. i guess I could pass the vkey here?